asset/asset.go

```diff
@@ -418,8 +433,45 @@ type GroupKey struct {
 	//   groupInternalKey + sha256(groupInternalKey || genesisOutPoint) * G
 	GroupPubKey btcec.PublicKey
 
-	// Sig is a signature over an asset's ID by `Key`.
-	Sig schnorr.Signature
+	// TapscriptRoot is the root of the Tapscript tree that commits to all
+	// script spend conditions for the group key. Instead of spending an
+	// asset, these scripts are used to define witnesses more complex than
+	// a Schnorr signature for reissuing assets.
+	TapscriptRoot [sha256.Size]byte
+
+	// Witness is a stack of witness elements that authorizes the membership
+	// of an asset in a particular asset group. The witness can be a single
+	// signature or a script from the tapscript tree committed to with the
+	// TapscriptRoot, and follows the witness rules in BIP-341.
+	Witness wire.TxWitness
+}
+
+// GroupKeyReveal is a type for representing the data used to derive the tweaked
+// key used to identify an asset group. The final tweaked key is the result of:
+// TapTweak(groupInternalKey, tapscriptRoot)
+type GroupKeyReveal struct {
+	// RawGroupKey is the public key that is tweaked twice to derive the
+	// final tweaked group key. The final tweaked key is the result of:
+	// groupInternalKey =  RawKey * sha256(assetID || RawKey) * G.
+	// GroupPubKey = TapTweak(groupInternalKey, TapscriptRoot)
+	RawKey SerializedKey
+
+	// TapscriptRoot is the root of the Tapscript tree that commits to all
+	// script spend conditions for the group key. Instead of spending an
+	// asset, these scripts are used to define witnesses more complex than
+	// a Schnorr signature for reissuing assets.
+	TapscriptRoot [sha256.Size]byte
+}
+
+// GroupPubKey returns the group public key derived from the group key reveal.
+func (g *GroupKeyReveal) GroupPubKey(assetID ID) (*btcec.PublicKey, error) {
+	rawKey, err := g.RawKey.ToPubKey()
+	if err != nil {
+		return nil, err
+	}
+
+	internalKey := input.TweakPubKeyWithTweak(rawKey, assetID[:])
```

Comment:
Linking this comment on using existing tweak methods:

https://github.com/lightningnetwork/lnd/pull/7898#issuecomment-1686536180